fitbenchmarking/parsing/parse_fitbenchmark_data.py

```diff
@@ -58,7 +58,7 @@ def __init__(self, fname):
 
         self._name = entries['name']
         self._equation = entries['function']
-        self._type = "FitBenchmark"
+        self._type = ((os.path.basename(__file__)).split('_')[1]).upper()
```

Comment:
I think you can remove self._type from here and in equivalent file for NIST by using suggestion here for getting module name in https://stackoverflow.com/questions/2020014/get-fully-qualified-class-name-of-an-object-in-python in func_def.py

Reply:
I will look into that.